x-pack/plugins/observability/public/pages/rules/index.tsx

```diff
@@ -250,13 +277,47 @@ export function RulesPage() {
         }}
       />
       <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiFieldSearch
+            fullWidth
+            isClearable
+            data-test-subj="ruleSearchField"
+            onChange={(e) => {
+              setInputText(e.target.value);
+              if (e.target.value === '') {
```

Comment:
@mgiota, The auto-refresh doesn't consider what is inside the search bar. The search is only triggered when the search bar is empty. Or when I hit enter in the second half of the screen recording.

Again, maybe this is the intended behavior? 
![Rules---Observability---Elastic](https://user-images.githubusercontent.com/6838659/159001704-a2dbb79a-046e-4c8b-acef-d157bce8ea05.gif)

Reply:
@fkanout Oh yeah good point. I didn't consider the scenario you describe at all. You are completely right, the search is triggered when the search bar is empty or when we hit enter

I did a quick test to the existing Stack Management > Rule page and the refresh doesn't consider what is in the search bar as well. (They don't have an autorefresh button there, that's why I did a quick test with the refresh button).

I can merge current PR as it is and I will create another ticket where we can implement this.